graylog2-server/src/main/java/org/graylog2/web/resources/AppConfigResource.java

```diff
@@ -43,14 +48,21 @@
     private final Configuration configuration;
     private final HttpConfiguration httpConfiguration;
     private final Engine templateEngine;
+    private final Map<String, String> pluginUISettings;
+    private final ObjectMapper objectMapper;
 
     @Inject
     public AppConfigResource(Configuration configuration,
                              HttpConfiguration httpConfiguration,
-                             Engine templateEngine) {
+                             Engine templateEngine,
+                             Set<PluginUISettingsProvider> settingsProviders,
+                             ObjectMapper objectMapper) {
         this.configuration = requireNonNull(configuration, "configuration");
         this.httpConfiguration = requireNonNull(httpConfiguration, "httpConfiguration");
         this.templateEngine = requireNonNull(templateEngine, "templateEngine");
+        this.pluginUISettings = requireNonNull(settingsProviders, "settingsProviders").stream()
+                .collect(Collectors.toMap(PluginUISettingsProvider::pluginSettingsKey, PluginUISettingsProvider::pluginSettingsJson));
```

Comment:
What happens if two plugins register with the same `pluginSettingsKey`? Does the collect fail or will it overwrite a previous key?

That said, do we need to think about namespacing for the keys? (e.g. with `PluginMetaData#getUniqueId?`)

Reply:
If we do not want to couple it to the unique id (which is by convention a fqcn, which is rather cryptic), we should at least use a `MapBinder` with the key being the key in the generated config.js. This would prevent silently overwriting the same key and fail at runtime instead if two plugins try to fill the same key.

Reply:
Need to give some thought to namespacing.  I agree that fqcn is probably too cryptic.  Maybe something similar to Java package naming like <your tld>.<your company/org>.<plugin name>.<specifier> such that the one I'm building for the Theme Customizer might be `org.graylog.themecustomizer.themeoverrides`

Reply:
Updated documentation on pluggable interface to include key naming conventions to avoid a collision.